index.js

```diff
@@ -181,34 +181,32 @@ class Ky {
 
 		this._options.headers = headers;
 
-		this._response = this._fetch();
+		const fn = async () => {
+			let response = await this._fetch();
 
-		for (const type of responseTypes) {
-			const isRetriableMethod = retryMethods.has(this._options.method.toLowerCase());
-			const fn = async () => {
-				if (this._retryCount > 0) {
-					this._response = this._fetch();
-				}
+			for (const hook of this._hooks.afterResponse) {
+				// eslint-disable-next-line no-await-in-loop
+				const modifiedResponse = await hook(response.clone());
 
-				let response = await this._response;
+				if (modifiedResponse instanceof Response) {
+					response = modifiedResponse;
+				}
+			}
 
-				for (const hook of this._hooks.afterResponse) {
-					// eslint-disable-next-line no-await-in-loop
-					const modifiedResponse = await hook(response.clone());
+			if (!response.ok && this._throwHttpErrors) {
+				throw new HTTPError(response);
+			}
 
-					if (modifiedResponse instanceof Response) {
-						response = modifiedResponse;
-					}
-				}
+			return response;
+		};
 
-				if (!response.ok && (isRetriableMethod || this._throwHttpErrors)) {
-					throw new HTTPError(response);
-				}
+		const isRetriableMethod = retryMethods.has(this._options.method.toLowerCase());
+		this._response = isRetriableMethod ? this._retry(fn) : fn();
```

Comment:
Unrelated to this PR, but it's been bothering me that `this._response` is saved on the instance despite only being used in the scope of the constructor. I'm pretty sure we can just do `const result =` instead of `this._response =`.

Reply:
Sure, I saw this too but maybe it was done like so for some reason.

@sindresorhus What do you think?

Reply:
👍👍

Reply:
`const result` or `this._response`?

Reply:
`const result`